resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/delegates/CookieHeaderDelegate.java

```diff
@@ -16,10 +18,40 @@ public Cookie fromString(String value) throws IllegalArgumentException
       return CookieParser.parseCookies(value).get(0);
    }
 
-   public String toString(Cookie value)
+   public String toString(Cookie cookie)
    {
-      StringBuffer buf = new StringBuffer();
-      ServerCookie.appendCookieValue(buf, 0, value.getName(), value.getValue(), value.getPath(), value.getDomain(), null, -1, false);

```

Comment:
JAX-RS seems to be a bit schizophrenic on the subject of cookies. We're talking about how to address this. Stay tuned.